source/material_model/rheology/visco_plastic.cc

```diff
@@ -0,0 +1,893 @@
+                                                           j, in.composition[i], current_cell,
+                                                           output_parameters.current_friction_angles[j],
+                                                           in.position[i]);
+
+
+            // Step 4: plastic yielding
+
+            // Determine if the pressure used in Drucker Prager plasticity will be capped at 0 (default).
+            // This may be necessary in models without gravity and the dynamic stresses are much higher
+            // than the lithostatic pressure.
+
+            double pressure_for_plasticity = in.pressure[i];
+            if (allow_negative_pressures_in_plasticity == false)
+              pressure_for_plasticity = std::max(in.pressure[i],0.0);
+
+            // Step 4a: calculate Drucker-Prager yield stress
+            const double yield_stress = drucker_prager_plasticity.compute_yield_stress(current_cohesion,
+                                                                                       output_parameters.current_friction_angles[j],
+                                                                                       pressure_for_plasticity,
+                                                                                       drucker_prager_parameters.max_yield_stress);
+
+            // Step 4b: select if yield viscosity is based on Drucker Prager or stress limiter rheology
+            double viscosity_yield = viscosity_pre_yield;
+            switch (yield_mechanism)
+              {
+                case stress_limiter:
+                {
+                  //Step 4b-1: always rescale the viscosity back to the yield surface
+                  const double viscosity_limiter = yield_stress / (2.0 * ref_strain_rate)
+                                                   * std::pow((edot_ii/ref_strain_rate),
+                                                              1./exponents_stress_limiter[j] - 1.0);
+                  viscosity_yield = 1. / ( 1./viscosity_limiter + 1./viscosity_pre_yield);
+                  break;
+                }
+                case drucker_prager:
+                {
+                  // Step 4b-2: if the current stress is greater than the yield stress,
+                  // rescale the viscosity back to yield surface
+                  // If this is the fault material and rate-and-state friction is used,
+                  // assume that we are always yielding
+                  //std::cout << "j is: " << j << ", while fault_material_index +1 : " << friction_options.fault_composition_index +1 << std::endl;
+                  //std::cout << "current_stress - yield_stress is: " << current_stress - yield_stress  << " in composition " << j<< std::endl << std::endl;
+                  //                   AssertThrow(volume_fractions.size() == in.composition[0].size()+1, ExcMessage("there is a field too much...."));
+
+                  if ((current_stress >= yield_stress) |
```

Comment:
```suggestion
                  if ((current_stress >= yield_stress) ||
```
You have to be careful with these logical symbols. It might compile, but `|` is only used for bitwise comparison, which is usually not what we do, generally `||` is needed!